File: apps/web/lib/api/cards.ts

import type { Card, NewCard } from "@/types/cards";
import { makeFetch } from "../fetch";
import { getApiUrl } from "../url";


export async function getCards() {
  try {
    const { data, error } = await makeFetch<Card[]>({
      url: getApiUrl("/v1/my/cards")
    })

    if (error) {
      throw error;
    }

    return data;

  } catch (error) {
    console.error(error)
    return null
  }
}

export async function createCard({ description, title, user_id }: NewCard) {
  try {
    const { data, error } = await makeFetch<Card>({
      url: getApiUrl("/v1/my/cards"),
      method: "POST",
      body: {
        description, title, user_id
      }
    })

    if (error) {
      throw error;
    }

    return data;

  } catch (error) {
    console.error(error)
    return null
  }
}

File: apps/api/internal/repositories/cards.go
package repositories

import (
	"github.com/yz13-dev/imc/api/internal/models"
	"gorm.io/gorm"
)

func GetCards(UserID string, db *gorm.DB) ([]models.Card, error) {
	var cards []models.Card
	err := db.Find(&cards, "user_id = ?", UserID).Error
	return cards, err
}

func CreateCard(UserID string, data *models.NewCard, db *gorm.DB) (models.Card, error) {
	var card = models.Card{
		UserID:      UserID,
		Title:       data.Title,
		Description: data.Description,
	}
	if err := db.Table("cards").Create(&card).Error; err != nil {
		return models.Card{}, err
	}
	return card, nil
}
